ui/src/map/layers/utils/alignment-layer-utils.ts

```diff
@@ -135,8 +139,9 @@ export function createAlignmentFeatures(
     );
 }
 
-function includes(selection: ItemCollections, alignment: AlignmentHeader): boolean {
-    switch (alignment.alignmentType) {
+function includes(selection: ItemCollections, alignment: LayoutAlignmentHeader): boolean {
+    const type = alignment.alignmentType;
```

Comment:
Nitpick: Tämä sijoitus on todennäköisesti turha koska se toimi aiemmallakin tavalla eikä tällä arvolla tehdä mitään muuta

Reply:
Joo, mutta TypeScript-kääntäjän mielestä ei ole. Tätä voi pitää yhtenä hyvänä syynä olla nyt ainakin jonkin verran varovainen monimutkaisempien tyyppien kanssa: Joskus täytyy tehdä muutoksia, joissa ei ole oikein mitään järkeä muuten kuin että ne auttavat kääntäjää näkemään jonkin selkeän asian.

Reply:
Näitä ekstramuuttujia on tullut itsekin välillä laitettua, mutta nyt kun tätä pohti hieman lisää, niin tämän näyttäisi kyllä saavan toimimaan ilmankin tuota lokaalia ekstramuuttujaa. Jos objektityypin rajaa never-tyypiksi asti, niin neveristä ei voi enää hakea kenttää, koska koko objektin instanssi on never, jolloin guardille voisi nakata koko varsinaisen objektin. Tulkinnan puolesta saattaa tosin olla hieman hähmeää miksi switchattua kenttää ei voida passata guardille. Esim: (ks. [TypeScript Playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAyg9gWwgFXNAvFA3lUkBcUA5AM6ITAAWAlgHYDmRUAvlAD7a5qGnlV30AogBsSEJswDcAKGkBjOLRLAoEAB6UAhgFdl1AG4QAspuByaDAOLbNAJwAmUTAAoA+oVoRDtgJQevELZOAHzY0lARuJS2cADuUJ7xgrYxts5EMJRw2sKOtHAqWoZQthCa5hCO-CRQCvbiPjJSsgBm2rRywNSKUJC2LXC2CACCnd20znTKmh0QhPBIqJA+YREksdRmlFCTSsAzchAAdHgQK1jhkXKaYsRkSPwMRPiXkVAARqWaANYyr7U3aC8B4WISicQvN4RT5lX6yN71Fo6YTASFQ0rAbS2WiqDQ6PSGExbATWOz2XbTWaNKAAehpUAA8t9LsxpKzpG0Ol0en0BkNRtzaAAmCn7WbzchLM6rKDrTbmHZTMWHE5oc7-a63YEUUHPf7Qr5wjWAu58UEiMR6qEfQ1-BEQJE5VH6koULE49RaXRdQmmcwkmwOUUHY6nal0qDJVIstlAA))

```
type SomeType = { type: 'something' } | { type: 'somethingElse' };

const exhaustiveMatchingGuard = (_: never): never => {
    throw new Error('Should not have reached this code');
};

function performAction(instance: SomeType) {
  switch (instance.type) {
    case 'something':
      break;

    case 'somethingElse':
      break;

    default:
      return exhaustiveMatchingGuard(instance); // Ok
  }
}

function performAction2(instance: SomeType) {
  switch (instance.type) {
    case 'something':
      break;

    case 'somethingElse':
      break;

    default:
      return exhaustiveMatchingGuard(instance.type); // Error
  }
}
```